core/src/main/java/com/digitalpebble/storm/crawler/parse/ParseResult.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * DigitalPebble licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.digitalpebble.storm.crawler.parse;
+
+import java.util.*;
+
+import com.digitalpebble.storm.crawler.Metadata;
+
+public class ParseResult implements Iterable<Map.Entry<String, ParseData>> {
+
+    private String URL;
```

Comment:
Doesn't have any real use for the moment, but I think of this attribute as a way of keeping a record of the original "parent" URL, the `ParseFilter` receive the original URL as a parameter, I can't think of a particular use case needing this, remove it then?